modules/402-ingress-nginx/hooks/get_ingress_controllers.go

```diff
@@ -144,6 +174,41 @@ func setInternalValues(input *go_hook.HookInput) error {
 			// controllerVersion should be absent if not specified explicitly
 			version = defaultControllerVersion // it's used only for metrics
 		}
+
+		inlet, found, err := unstructured.NestedString(controller.Spec, "inlet")
+		if err != nil {
+			return fmt.Errorf("cannot get inlet type from ingress controller spec: %v", err)
+		}
+		if found && inlet == "L2LoadBalancer" {
+			var nodeSelectorLabels = labels.Everything()
+			if nodeSelector, exists := controller.Spec["nodeSelector"]; exists {
+				nodeSelectorMapInterface := nodeSelector.(map[string]interface{})
+				nodeSelectorMapStr := make(map[string]string)
+				for key, value := range nodeSelectorMapInterface {
+					strValue := value.(string)
+					nodeSelectorMapStr[key] = strValue
+				}
+				nodeSelectorLabels = labels.Set(nodeSelectorMapStr).AsSelector()
+			}
```

Comment:
```suggestion
			}
			nodeSelectorLabels = labels.Set(nodeSelectorMapStr).AsSelector()
```
Maybe it is the solution for empty nodeSelector?